### PR TITLE
test: add more layer e2e tests, fix existing ones

### PR DESCRIPTION
**Pull request by attilah**
test: add more layer e2e tests, fix existing ones

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-core/src/utils/index.ts [hunk 1]
```diff
@@ -33,58 +34,95 @@ export function deleteAmplifyDir(root: string) {
...
+export function loadFunctionTestFile(fileName: string) {
+  const functionPath = getTestFileNamePath(fileName);
+  return fs.readFileSync(functionPath, 'utf-8').toString();
 }
 
-export function overrideFunctionCode(root: string, name: string, fileName: string) {
-  let indexPath = path.join(getPathToFunction(root, name), 'src', 'index.js');
-  let functionPath = path.join(__dirname, '..', '..', '..', 'amplify-e2e-tests', 'functions', fileName);
-  fs.copySync(functionPath, indexPath);
+export function addNodeJSDependencies(root: string, functionName: string, dependencies: string[]) {
```
Comment (jhockett) [inline packages/amplify-e2e-core/src/utils/index.ts:42]: nit: function name can just be `addNodeDependencies` to match naming convention used below

File: packages/amplify-e2e-tests/functions/case-function-for-layer.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const fs = require('fs');
```
Comment (jhockett) [inline packages/amplify-e2e-tests/functions/case-function-for-layer.js:1]: can this import be removed?

**Review by jhockett — APPROVED** [review]
LGTM

File: packages/amplify-e2e-core/src/utils/index.ts (hunk 1 above)
Comment (attilah) [reply packages/amplify-e2e-core/src/utils/index.ts:42]: I just moved it, but will rename :)

**Pull request merged**